include/openPMD/IO/ADIOS/ADIOS1Auxiliary.hpp

```diff
@@ -87,24 +90,86 @@ getBP1DataType(Datatype dtype)
         case DT::VEC_UCHAR:
         case DT::BOOL:
             return adios_unsigned_byte;
-        case DT::INT16:
-        case DT::VEC_INT16:
-            return adios_short;
-        case DT::INT32:
-        case DT::VEC_INT32:
-            return adios_integer;
-        case DT::INT64:
-        case DT::VEC_INT64:
-            return adios_long;
-        case DT::UINT16:
-        case DT::VEC_UINT16:
-            return adios_unsigned_short;
-        case DT::UINT32:
-        case DT::VEC_UINT32:
-            return adios_unsigned_integer;
-        case DT::UINT64:
-        case DT::VEC_UINT64:
-            return adios_unsigned_long;
+        case DT::SHORT:
+        case DT::VEC_SHORT:
+            if( sizeof(short) == 2u )
+                return adios_short;
+            else if( sizeof(short) == 4u )
+                return adios_integer;
+            else if( sizeof(long) == 8u )
+                return adios_long;
+            else
+                throw unsupported_data_error("No native equivalent for Datatype::SHORT found.");
+        case DT::INT:
+        case DT::VEC_INT:
+            if( sizeof(int) == 2u )
+                return adios_short;
+            else if( sizeof(int) == 4u )
+                return adios_integer;
+            else if( sizeof(int) == 8u )
+                return adios_long;
+            else
+                throw unsupported_data_error("No native equivalent for Datatype::INT found.");
+        case DT::LONG:
+        case DT::VEC_LONG:
+            if( sizeof(long) == 2u )
+                return adios_short;
+            else if( sizeof(long) == 4u )
+                return adios_integer;
+            else if( sizeof(long) == 8u )
+                return adios_long;
+            else
+                throw unsupported_data_error("No native equivalent for Datatype::LONG found.");
```

Comment:
`long` is at least 32 Bit, making the
```cpp
if( sizeof(long) == 2u )
```
branch dead code.
[C++ fundamental types](https://en.cppreference.com/w/cpp/language/types)

Reply:
But a Byte is not necessarily 8 Byte ;-)

Reply:
I just want to mention that I can't see portability in ADIOS itself for bytes other than 8 bit, but maybe I am missing it.

Reply:
I would leave this in, just because it makes it easier to read and the compile time dead branches should be optimized out by a good compiler, no? (Or does it throw warnings somewhere?)

Reply:
> I can't see portability in ADIOS1 itself for bytes other than 8 bit

No automatic way, they assume the user to do it by hand (please don't).

> I would leave this in

There's a point where software is generalized too much. This is not there yet, but close. Leave it in, I don't blame you for copy pasting and saving time. ;)

Reply:
> But a Byte is not necessarily 8 Bit ;-)

This logic then only works for 
```Python
CHAR_BIT in (4, 8, 16, 32)
```
btw ;)